### PR TITLE
Introduce Deneb to remove EIP-4844 references

### PR DESCRIPTION
**Pull request by rolfyone**
Consensus-spec references may need to be fixed once deneb is renamed in that project.

related to https://github.com/ethereum/consensus-specs/issues/3207

### CODE REVIEW

**Review by mcdee — APPROVED** [review]

**Comment by jimmygchen** [comment]
Hi @rolfyone thanks for this - is this PR ready to be merged?

File: types/capella/block.yaml [hunk 1]
```diff
@@ -73,10 +73,6 @@ Capella:
         properties:
           execution_payload_header:
             $ref: './execution_payload.yaml#/Capella/ExecutionPayloadHeader'
-          bls_to_execution_changes:
```
Comment (Inphi) [inline types/capella/block.yaml:76]: Why is this removed? Is this related to EIP-4844/deneb?

**Comment by rolfyone** [comment]
> Hi @rolfyone thanks for this - is this PR ready to be merged?

won't be able to merge it until we've done the capella release...

File: types/capella/block.yaml (hunk 1 above)
Comment (rolfyone) [reply types/capella/block.yaml:76]: this seems like maybe a branch out of date issue ,but does look weird...

Comment (rolfyone) [reply types/capella/block.yaml:76]: ok so this looks like a cleanup, it is in BeaconBlockBodyCommon, so it gets pulled into here when the `allOf` includes `BeaconBlockBodyCommon`

**Comment by rolfyone** [comment]
Trying to get the capella release so this can move. I've asked in API's channel. hopefully this isn't completely stale!

**Comment by rolfyone** [comment]
> any updates on this? Are we still waiting on a Capella release?

this is now unblocked, awaiting approval to merge to dev.

File: types/deneb/block.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  BeaconBlockBody:
+    allOf:
+      - $ref: '#/Deneb/BeaconBlockBodyCommon'
+      - type: object
+        properties:
+          execution_payload:
+            $ref: './execution_payload.yaml#/Deneb/ExecutionPayload'
+
+  BeaconBlock:
+    description: "The [`BeaconBlock`](https://github.com/ethereum/consensus-specs/blob/master/specs/Deneb/beacon-chain.md#beaconblock) object from the CL Deneb spec."
```
Comment (mcdee) [inline types/deneb/block.yaml:55]: URL should be lower-case D in Deneb for it to resolve.

File: types/deneb/state.yaml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        type: array
+        description: "Fixed length of 8192 items"
+        items:
+          $ref: '../primitive.yaml#/Root'
+      state_roots:
+        type: array
+        description: "Fixed length of 8192 items"
+        items:
+          $ref: '../primitive.yaml#/Root'
+      historical_roots:
```
Comment (mcdee) [inline types/deneb/state.yaml:26]: Is this now a 0-length array in the spec, given that it has been superseded by `historical_summaries`?  Also, we're missing `historical_summaries` in this definition.

**Review by mcdee — CHANGES_REQUESTED** [review]

File: types/deneb/state.yaml (hunk 3 above)
Comment (rolfyone) [reply types/deneb/state.yaml:26]: It looks like we're missing historical_summaries in the 2.4.0 release from capella, so that's a gap I might raise an issue on...  This is consistent with the Capella definition in master, so we need to fix capella and deneb both.

Comment (rolfyone) [reply types/deneb/state.yaml:26]: https://github.com/ethereum/consensus-specs/blob/dev/specs/capella/beacon-chain.md#beaconstate
I'm not sure it's a 0 length, but whatever it ends up, it does look like tech debt...

Comment (rolfyone) [reply types/deneb/state.yaml:26]: ok, was easy enough for capella, brought in change and moved to deneb also.

**Review by mcdee — APPROVED** [review]

**Review by michaelsproul — APPROVED** [review]
